p2p/state.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package p2p
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+)
+
+type BlockState interface {
+	AddBlock(types.Block) error
+	LatestHeader() *types.Header
+}
+
+type StorageState interface {
+	StorageRoot() (common.Hash, error)
+}
+
+type MockBlockState struct{}
+
+func (mbs *MockBlockState) AddBlock(types.Block) error {
+	return nil
+}
+
+func (mbs *MockBlockState) LatestHeader() *types.Header {
```

Comment:
maybe put the MockBlockState in the tests?

Reply:
yes, good idea, prod code mixed with mocks is not a good practice.

Reply:
will do